Assets/Scripts/Robo.cs

```diff
@@ -132,4 +133,11 @@ private IEnumerator Iframes()
 
         StartCoroutine(Iframes());
     }
+
+    
+    private IEnumerator GameOverDebug(){
+        yield return new WaitForSeconds(10);
+        playerInputs.Player.fire.performed -= FireDuncan;
```

Comment:
We should abstract this removal of `FireDuncan` into a general `OnDeath` function -- which I believe is also the strategy used in another PR. This way, we can aggregate these actions easily.

Reply:
I think I may have (offhandedly?) mentioned this before in a voice-chat discussion, but in instances like these where the future intent/implementation is clear, it is fine to have partial/redundant functions that will be saturated and made _more useful/relevent_ later.

In this case, that means have a very simple one-line `OnDeath` function that we later populate further.

There's definitely a balance to strike between "future-proof everything" and "retrofit everything" but I think in this case we can safely aim to futureproof since we've got confidence that such a fixture would be helpful. The main benefit, imo, is to make merging and future review easier.

Reply:
Yeah, don't get me wrong, I agree with this take in this instance. It's an easy edit, I'll push the change later today. The only reason why I provided pushback is because I don't entirely see the difference between this situation and, say, cleaning up typos in previously written, unrelated comments in the middle of a pull request. Like, in both cases, I use one branch to partially complete an unrelated task (implementing player death vs fixing deviations from the MUGA). My guess at the difference is that here I'm writing new code rather than editing code from a previous task, but even within that there's still obviously ambiguity, like I *could* have written up an abstract enemy class when mentioned in a previous code review and just not changed our current enemy to extend it yet, and that would similarly make some future task easier by getting some of the work out of the way now without making any edits to previously written code, just without any direct bearing on the task in which it was mentioned.

There's also obviously the question of how extensively the OnDeath() task should be implemented in this task. Like, obviously we need to empty the fire.performed InputAction here so we can load a scene without error, which is within the task's scope. We could also start the game over display coroutine from within OnDeath since that's also something we'll want in the OnDeath method later that we need to do in the debug coroutine anyway. What about adding disabling of the player's move action, however? Strictly speaking, it is completely unnecessary for this current task, but then again, so is separating out those two lines into an OnDeath method in the first place, and in both cases it'll make a future pull request's review easier while making this one marginally harder. What about implementing GameEvents and GameEventListeners so we can add a player death event trigger in the OnDeath method that other entities can listen for? It's a much more extreme case than just making an OnDeath method, but similarly it's just adding extra code in this pull request that'll make a future pull request review easier. It's worth noting that this pull request isn't a super large one, and I know ahead of time that implementing player death *will* be a large one, so since the difficulty of reviewing a pull request scales quadratically with respect to quantity written (as you verbally said at one point), it simply makes sense for me to take some large amount of code from that future pull request and arbitrarily implement it here to make future you have an easier time, given that we're already willing to expand the scope of this pull request for futureproofing purposes.

The reason why I'm going into this is because, well, I wasn't asking these questions before because I thought the expectation was clear, avoid starting implementation of a future task unless it's necessary for the current task. This keeps tasks and pull requests similarly... while not atomic like a commit, similarly self-contained in scope. This expectation appears to have been wrong, and I feel very unsure of any rules of thumb to differentiate the cases at the moment. As such, I feel most comfortable requesting that you attempt to clarify your stance in a general case or at least several specific particular cases that have already come up and will likely come up in the future in the MUGA, because it's just not going to be feasible if we ever get more programmers on the project for you to individually tweak every person's implicit metric for pull request scope within these broad bounds. It'll just save time on the manager's end in the long-run if we can point at a rule of thumb in most cases and say "hey, this time it's something it makes sense to separate out, check the MUGA's pull request section."

I'd do it myself, but I think the past 3 paragraphs show that I don't know yet what that section would need to say.

Reply:
Pushed! Going to keep this open for now, see above request. Let me know if you want me to make it a task on Notion so you can keep track of it more easily (you're busy, I understand).